plugins/inputs/ecs/client_test.go

```diff
@@ -106,6 +105,8 @@ func TestEcsClient_Task(t *testing.T) {
 			name: "happy",
 			client: mockDo{
 				do: func() (*http.Response, error) {
+					rc, err := os.Open("testdata/metadata.golden")
+					require.NoError(t, err)
```

Comment:
I would do
```suggestion
					if err != nil {
						return nil, err
					}
```
Same below.